src/studio/src/designer/frontend/app-development/features/dataModeling/dataModelingSlice.ts

```diff
@@ -63,6 +63,24 @@ const dataModelingSlice = createSlice({
       const { modelName } = action.payload;
       state.modelName = modelName;
     },
+    newDataModel(state, action) {
+      const { modelName } = action.payload
+      state.modelName = modelName;
+      state.error = null;
+      state.schema = {};
+    },
+    deleteDataModel(state, action) {
```

Comment:
General comment on these methods.
This file was the first one that was migrated to redux toolkit, and as such is not the best example since it was a bit of an experiment 😅 Some things that can be updated in the new code you have added:
- If the action parameter is not used in the reducer method, it can be removed.
- If the action parameter is in use, specify the type as `PayloadAction<ISomeInterface>` where the interface specifies the content of the payload of the action. See f.ex. `features/handleMergeConflict/handleMergeConflictSlice` for an example of what I mean.